core/config/config.go

```diff
@@ -85,6 +85,12 @@ type Configuration struct {
 	HTTPClient            *http.Client
 	RetryOptions          *clients.RetryConfig
 
+	// If != nil, a goroutine will be launched that will refresh the service account's access token when it's close to being expired.
+	// The goroutine is killed whenever this context is cancelled.
+	//
+	// Only has effect for key flow
+	TokenRefreshInBackgroundContext context.Context
```

Comment:
Although I understand it, the name is not very user-friendly. I wonder if context is even necessary
Maybe `BackgroundTokenRefresh` or  `BackgroundTokenRefreshContext` ?

Reply:
I meant having `Context` in the name. But yeah let's keep it for clarity